core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -864,6 +864,178 @@ ThreadVectorRange(const TeamMemberType&, const iType1& arg_begin,
 
 namespace Impl {
 
+enum class TeamMDRangeLastNestLevel : bool { NotLastNestLevel, LastNestLevel };
+enum class TeamMDRangeParThread : bool { NotParThread, ParThread };
+enum class TeamMDRangeParVector : bool { NotParVector, ParVector };
+enum class TeamMDRangeThreadAndVector : bool { NotBoth, Both };
+
+struct NoReductionTag {
+  explicit NoReductionTag() = default;
+};
+
+static KOKKOS_IMPL_DEVICE_FUNCTION NoReductionTag no_reduction_tag{};
```

Comment:
It doesn't make sense to me to use `KOKKOS_IMPL_DEVICE_FUNCTION` here just because it does the right thing.
`SYCL`'s support for global device variables is pretty immature and so I would be happy if we could avoid doing this in the first place. If we must, the attribute should have a different name.

Reply:
Changed in [f5f24e3](https://github.com/kokkos/kokkos/pull/5238/commits/f5f24e33bb5c604fb0cbc471850d201f86ca308c).

Removed this global device variable. Now constructs it inline.